pkg/controller/miganalytic/miganalytics_controller.go

```diff
@@ -0,0 +1,582 @@
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	MiBSuffix = "Mi"
+	MiB       = 1048576
+)
+
+var log = logging.WithName("analytics")
+
+// Add creates a new MigAnalytic Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileMigAnalytic{Client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("miganalytic-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	// Watch for changes to MigAnalytic
+	err = c.Watch(
+		&source.Kind{Type: &migapi.MigAnalytic{}},
+		&handler.EnqueueRequestForObject{},
+		&AnalyticPredicate{})
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileMigAnalytic{}
+
+// ReconcileMigAnalytic reconciles a MigAnalytic object
+type ReconcileMigAnalytic struct {
+	client.Client
+	scheme *runtime.Scheme
+}
+
+func (r *ReconcileMigAnalytic) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	var err error
+	log.Reset()
+
+	// Fetch the MigAnalytic instance
+	analytic := &migapi.MigAnalytic{
+		Spec: migapi.MigAnalyticSpec{
+			MaxListImages: 25,
+		},
+	}
+
+	err = r.Get(context.TODO(), request.NamespacedName, analytic)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		log.Trace(err)
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	// Report reconcile error.
+	defer func() {
+		if err == nil || errors.IsConflict(err) {
+			return
+		}
+		analytic.Status.SetReconcileFailed(err)
+		err := r.Update(context.TODO(), analytic)
+		if err != nil {
+			log.Trace(err)
+			return
+		}
+	}()
+
+	// Begin staging conditions.
+	analytic.Status.BeginStagingConditions()
+
+	// Add Labels and Annoations
+	analytic.ObjectMeta.Labels = map[string]string{"migplan": analytic.Spec.Plan.Name}
+	analytic.ObjectMeta.Annotations = map[string]string{"migplan": analytic.Spec.Plan.Name}
+
+	// Validations.
+	err = r.validate(analytic)
+	if err != nil {
+		log.Trace(err)
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	// Update Validation Status
+	err = r.Update(context.TODO(), analytic)
+	if err != nil {
+		log.Trace(err)
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	// Analyze
+	if !analytic.Status.HasBlockerCondition() {
+		err = r.analyze(analytic)
+	} else {
+		log.Info("Requeuing MigAnalytic, associated plan not ready.", "miganalytic", analytic.Name)
```

Comment:
I suggest:

1. Removing this log statement. We only log phase change and errors.
2. Moving this info onto a new MigAnalytic status condition and use "postponed" language 

```
# Example from MigMigration
	// Postpone
	requeueAfter := time.Second * 10
	for position, uid := range pending {
		if uid == migration.UID {
			requeueAfter = time.Second * time.Duration(position*10)
			break
		}
	}
	migration.Status.SetCondition(migapi.Condition{
		Type:     Postponed,
		Status:   True,
		Category: Critical,
		Message:  fmt.Sprintf(PostponedMessage, requeueAfter/time.Second),
	})
```